src/bootstrap/native.rs

```diff
@@ -175,6 +174,12 @@ impl Step for Llvm {
             }
         }
 
+        if builder.config.lld_enabled {
+            cfg.define("LLVM_ENABLE_ZLIB", "ON");
```

Comment:
It's not clear to me if this setting has repercussions beyond LLD.

Reply:
Can you think of a reason we wouldn't want to support zlib even if lld isn't enabled? I imagine it's more broadly useful and not measurably slower to compile.

Reply:
I looked at the git history and couldn't find a reason for it being off. It may just have been a case of "we don't use it, so why enabled it?"